models/users.js

```diff
@@ -715,6 +715,27 @@ const getUsersByRole = async (role) => {
   }
 };
 
+const generateUniqueUsername = async (firstname, lastname) => {
```

Comment:
I cannot find any bugs in the code snippet you have provided. The code seems to be working as intended.

The function generateUniqueUsername() takes two parameters: the user's first name and last name. It first checks if there is already a user with the same first name and last name in the database. If there is, it increments the user count by 1 and appends that number to the username. Otherwise, it simply returns the username as-is.

The code snippet you have provided does not have any errors. However, there are some potential improvements that could be made. For example, the code could be made more efficient by using a parameterized query instead of a hardcoded query. Additionally, the code could be made more robust by handling errors more gracefully.

Here is an example of how the code could be improved:

const generateUniqueUsername = async (firstname, lastname) => {
  try {
    const query = userModel.where({
      first_name: firstname,
      last_name: lastname,
    });
    const count = await query.count();
    const initialUsername = `${firstname}-${lastname}`;
    const uniqueUsername = count === 0 ? initialUsername : `${initialUsername}-${count + 1}`;
    return uniqueUsername;
  } catch (err) {
    logger.error(`Error while generating unique username: ${err.message}`);
    throw err;
  }
};

Reply:
Hi tejas, the code snippet you provided is not actually running and its give error
then I research and found firestore does not support parameterized query in the way you described and actually there is no concept of parameterized query in firestore (both hardcoded and parameterized query idea same in firestore)

I want to achieve this
RFC: https://github.com/Real-Dev-Squad/website-backend/issues/1407
<img width="785" alt="image" src="https://github.com/Real-Dev-Squad/website-backend/assets/107163260/ef6f54a3-7d06-44a7-b291-03827f5d81b4">

To achieve above 

I found code directly from firestore docs- https://firebase.google.com/docs/firestore/query-data/aggregation-queries
<img width="665" alt="image" src="https://github.com/Real-Dev-Squad/website-backend/assets/107163260/76a0c8f2-31a9-446b-80dd-1f15228b281c">

Firestore has a billing optimization that allows you to be charged for one document read operation if your count() operation matches between 0 and 1000 index entries.